### ISSUE TITLE
AWS S3: uploadMultipart* with S3Headers fails

### ISSUE THREAD

**Issue by wi101** [akka/alpakka#2906]
### Versions used 

Alpakka version: 4.0.0
Akka version: 2.6.20

### Expected Behavior

`S3.multipartUpload` and `S3.multipartUploadWithHeaders` with S3Headers should succeed as before in v3.0.4

### Actual Behavior

When calling `S3.multipartUpload` or `S3.multipartUploadWithHeaders` with S3Headers there is an error.
```
akka.stream.alpakka.s3.S3Exception: The specified header is not valid in this context
```

When I tried to call `S3.multipartUploadWithHeaders` with an empty header it worked.
An assumption: this change in completionSink might be the reason: https://github.com/akka/alpakka/pull/2844

**Comment by sikinmettugi** [comment]
Hi, I also faced the same issue after updating alpakka-s3 from 3.0.1 to 4.0.0. From my side the current dependency versions are:
akka 2.6.19
alpakka 4.0.0
akka-http 10.2.9 (if this helps)

**Comment by hagerf** [comment]
Hi, just wanted to add that this is a breaking change also for us, we cannot use Alpakka 4.0.0 until this is fixed :(

**Comment by mvillafuertem** [comment]
Hi there, I also faced the same issue. Any workaround?

**Comment by cptwunderlich** [comment]
Alpakka version 5.0 is also affected.

**Comment by gipeshka** [comment]
The issue is apparently caused by sending metadata/custom headers with the multipart upload completion request.

**Comment by sebastian-alfers** [comment]
We published `5.0.0+17-5228417f-SNAPSHOT` in case anyone would like try out that [snapshot](https://doc.akka.io/docs/alpakka/current/other-docs/snapshots.html).

**Comment by hagerf** [comment]
Thanks @sebastian-alfers.
Would be great to test, but since the final version is going to be released in version >5.0.0 which is covered by BSL (I presume), I don't think we'll be able to use it in our company sadly. Or will some patch of 4.0.0 include this?

**Comment by sebastian-alfers** [comment]
I am sorry to hear that, but currently we do not plan to backport this fix.
